src/core/patterns.js

```diff
@@ -59,3 +59,61 @@ foam.CLASS({
     }
   ]
 });
+
+/** Causes an class to pool its instances. create() will pull from the pool,
+ and destroy() will return instances to the pool. Object pools can be found
+ in <code>foam.__objectPools__</code>. */
+foam.CLASS({
+  package: 'foam.pattern',
+  name: 'Pooled',
+  axioms: [ foam.pattern.Singleton.create() ],
+
+  methods: [
+    function installInClass(cls) {
+      // Keeping the object pools in an accessible location allows them
+      // to be cleared out.
+      if ( ! foam.__objectPools__ ) { foam.__objectPools__ = {}; }
```

Comment:
The only reason to keep them in one place is so you could wipe out all retained pools at once. Otherwise they would be better placed in the class.

An alternative would be to make a list of pooled classes in a shared location (and maybe it's optional) so you could iterate and clear them if necessary.